wolfcrypt/src/asn.c

```diff
@@ -13678,9 +13678,12 @@ static int MatchBaseName(int type, const char* name, int nameSz,
 {
     if (base == NULL || baseSz <= 0 || name == NULL || nameSz <= 0 ||
             name[0] == '.' || nameSz < baseSz ||
-            (type != ASN_RFC822_TYPE && type != ASN_DNS_TYPE))
+            (type != ASN_RFC822_TYPE && type != ASN_DNS_TYPE && type != ASN_DIR_TYPE))
```

Comment:
Can you reduce the maximum line length here (80 chars) and also add braces {} since it is a large logic block.

Reply:
done